antora/README.md

```diff
@@ -1,29 +1,35 @@
 
-This directory contains the configuration needed by [Antora][antora] to render
-the content of the `docs` directory and documentation sources from other
-directories as HTML.
+These are the sources for building the https://enterprisecontract.dev website.
+The website uses [Hugo](https://gohugo.io/) and
+[Antora][https://docs.antora.org/] to build different sections of the website.
 
-Run `npm run build` to generate the rendered HTML files in the `public`
-directory.
+## Live reload preview
 
-## Local development
+To run the preview of the website as changes are made to files within this
+repository run `make preview`. Note that the `preview` target also invokes
+`npm ci` in the `antora` directory which will reinstall and remove linked
+packages needed for development, running `npm run dev:setup` should restore the
+links and help with debugging. More on this below.
 
-Antora supports building from local git clone of the repository and with
-uncommited changes. This is done by changing the `content/sources` in
-`antora-playbook.yml` to point to the local directory for the `url` and `HEAD`
-for `branches.
+## Development
 
-With `hack/local-build.sh` script performs this by creating a temporary playbook
-file pointing to the local clones of the mentioned repositories if those
-repositories are located in the `../<repository>` for a
-`https://github.com/<org>/<repository>.git` git URL. For example, it is expected
-that the `ec-policies` repository is located at `../ec-policies`. If the local
-repository is on a different path create a symlink pointing from
-`../<repository>` to that path.
+The website makes makes use of
+[Antora Extensions](https://docs.antora.org/antora/latest/extend/extensions/),
+there are several extensions defined as development dependencies in
+`antora/package.json` and configured in `antora/antora-playbook.yml`.
 
-The `hack/local-live.sh` script calls `hack/local-build.sh` for any changes to
-files in the locally cloned repositories. Useful to leave in the background and
-have the generated HTML site continuously updated.
+To make changes locally to the extensions run `npm run dev:setup` script in the
+`antora` directory. The script assumes that the local clones of
+[ec-cli](https://github.com/enterprise-contract/ec-cli/) and
+[ec-policies](https://github.com/enterprise-contract/ec-policies/) repositories
+are present in the directory above this one. The outcome of running the
+`dev:setup` script is that local copies of the NPM packages that comprise the
+extensions are [linked](https://docs.npmjs.com/cli/v6/commands/npm-link), so
+instead of the versions released to npmjs.com, local versions with any local
+changes will be used.
 
-[antora]: https://docs.antora.org/
+Opening the `enterprise-contract.github.io.code-workspace` workspace file within
+VSCode will load the Node projects of the same extensions and the `Antora Build`
+launch configuration in `.vscode/launch.json` allows running the debugger
```

Comment:
I'm trying this, but my breakpoints on the `index.js` of each extension is ignored. I did run `npm run dev:setup` prior.

Reply:
Do this:
```shell
$ git rev-parse HEAD
6893a5fd05e7c4d193ec04c175e443eca35c0833
$ git clean -dfx
$ cd antora
$ npm run dev:setup

> @enterprise-contract/website@0.0.0 dev:setup
> npm run dev:links && npm link @enterprise-contract/ec-policies-antora-extension @enterprise-contract/reference-antora-extension @enterprise-contract/rego-antora-extension @enterprise-contract/tekton-task-antora-extension


> @enterprise-contract/website@0.0.0 dev:links
> (cd ../../ec-policies/antora/ec-policies-antora-extension && npm link) && (cd ../../ec-cli/reference-antora-extension && npm link) && (cd ../../ec-cli/rego-antora-extension && npm link) && (cd ../../ec-cli/tekton-task-antora-extension && npm link)


up to date, audited 3 packages in 658ms

found 0 vulnerabilities
Reshimming asdf nodejs...

up to date, audited 3 packages in 672ms

found 0 vulnerabilities
Reshimming asdf nodejs...

up to date, audited 3 packages in 663ms

found 0 vulnerabilities
Reshimming asdf nodejs...

up to date, audited 3 packages in 612ms

found 0 vulnerabilities
Reshimming asdf nodejs...

removed 1 package, changed 4 packages, and audited 241 packages in 721ms

34 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

$ ls -l node_modules/@enterprise-contract # all of these should be symbolic links
total 16
lrwxrwxrwx. 1 zregvart zregvart 59 Mar 26 10:29 ec-policies-antora-extension -> ../../../../ec-policies/antora/ec-policies-antora-extension
lrwxrwxrwx. 1 zregvart zregvart 45 Mar 26 10:29 reference-antora-extension -> ../../../../ec-cli/reference-antora-extension
lrwxrwxrwx. 1 zregvart zregvart 40 Mar 26 10:29 rego-antora-extension -> ../../../../ec-cli/rego-antora-extension
lrwxrwxrwx. 1 zregvart zregvart 47 Mar 26 10:29 tekton-task-antora-extension -> ../../../../ec-cli/tekton-task-antora-extension

$ cd ..
$ code enterprise-contract.github.io.code-workspace
```

Now set a breakpoint in one of the extension's `index.js` files. Go to the Run and Debug tab and run the `Antora Build (enterprise-contract.github.io)` launch configuration.

Do **not** run any other `npm` or `make` commands, they will run `npm ci` which will remove the symlinks created by the `dev:setup` script.